File: 卡码/代码随想录/JS/栈与队列/03-逆波兰表达式/LeetCode_0150.js
//150. 逆波兰表达式求值
const evalRPN = function(tokens) {
  const nums=new Array();
  let n1,n2;
  for(let token of tokens){
    if(token=="+"||token=="-"||token=="*"||token=="/"){
      n2=nums.pop();
      n1=nums.pop();
    }
    switch(token){
      case "+":
        nums.push(n1+n2);
        break;
      case "-":
        nums.push(n1-n2);
        break;
      case "*":
        nums.push(n1*n2);
        break;
      case "/":
        let num=n1/n2;
        nums.push(num>0 ? Math.floor(num) : Math.ceil(num));
        break;
      default:
        nums.push(Number(token));
    }
  }
  return nums[nums.length-1];
};

File: 卡码/代码随想录/C++/栈与队列/03-逆波兰表达式/LeetCode_0150.cpp
//150. 逆波兰表达式求值
#include <vector>
#include <string>
using namespace std;
class Solution {
public:
  int evalRPN(vector<string>& tokens) {
    stack<int> nums;
    int n1,n2;
    for(string token:tokens){
      if(token=="+"||token=="-"||token=="*"||token=="/"){
        n2=nums.top();
        nums.pop();
        n1=nums.top();
        nums.pop();
        if(token=="+") nums.push(n1+n2);
        else if(token=="-") nums.push(n1-n2);
        else if(token=="*") nums.push(n1*n2);
        else nums.push(n1/n2);
      }else{
        nums.push(stoi(token));
      }
    }
    return nums.top();
  }
};
